tests/lib/devicetree/api/src/main.c

```diff
@@ -1537,6 +1542,9 @@ static void test_enums_required_false(void)
 	zassert_equal(DT_ENUM_IDX_OR(DT_NODELABEL(test_enum_int_default_1),
 				     val, 4),
 		      4, "");
+	/* DT_INST_ENUM_IDX_OR (using vnd,enum-required-false-holder) */
+	zassert_equal(DT_INST_ENUM_IDX_OR(0, val, 2), 1, "");
+	zassert_equal(DT_INST_ENUM_IDX_OR(1, val, 2), 2, "");
```

Comment:
It is never safe to make assumptions about the order of instances. Please read https://docs.zephyrproject.org/latest/reference/devicetree/api.html#c.DT_INST and rework this test.

Reply:
Well, "never" is a bit of a strong word, but this is unsafe :)